stl/inc/memory

```diff
@@ -2685,6 +2714,13 @@ void _Uninitialized_fill_multidimensional_n_al(_Ty* const _Out, const size_t _Si
     } else if constexpr (_Fill_memset_is_safe<_Ty*, _Ty> && _Uses_default_construct<_Alloc, _Ty*, const _Ty&>::value) {
         _Fill_memset(_Out, _Val, _Size);
     } else {
+        if constexpr (_Fill_zero_memset_is_safe<_Ty*,
+                          _Ty> && _Uses_default_construct<_Alloc, _Ty*, const _Ty&>::value) {
+            if (_Is_scalar_zero(_Val)) {

```

Comment:
Ditto "Does this new special case have test coverage?"

Reply:
Ditto "AFAIK this function is not tested anywhere so I'm unsure what to do"

Reply:
Done